src/cascadia/TerminalSettingsModel/AIConfig.cpp

```diff
@@ -126,6 +126,13 @@ void AIConfig::ActiveProvider(const LLMProvider& provider)
 
 winrt::hstring AIConfig::_RetrieveCredential(const std::wstring_view credential)
 {
+    const auto credentialData = credential.data();
+    // first check our cache
+    if (_credentialCache.contains(credentialData))

```

Comment:
We only ever call this function with the `static constexpr std::wstring_view`s defined at the top of this file though, so it should be okay right?

Reply:
Yeah, I thought so and so I approved the PR. It's still a good idea to not use `.data()` because null-terminated strings are kind of slow (because of the unknown length) and error prone (what if there isn't a null-terminator after all?). I just realized you could change the parameter to `wil::zwstring_view`. That gives it a `.c_str()` method and the absolute guarantee that it will be null-terminated. There are many options... 😅